perceval/algorithm/tomography/tomography_mle.py

```diff
@@ -0,0 +1,400 @@
+        preparation_indices = [PauliEigenStateType.Zm]  * self._nqubit
+        # Input Preparation fixed to |0> for state tomography
+
+        for val in measurement_indices:
+            if PauliType.Z in val:
+                continue
+            f += self._collect_data(preparation_indices, state_meas_indices=val)
+
+        return f
+
+    def _log_likelihood_func(self, rho: np.ndarray) -> float:
+        # Log-likelihood function of the POVM to minimize
+
+        f = self._f
+        P = self._povm_operator()
+
+        x = 0
+        for k in range(len(f)):
+            if np.trace(np.dot(rho, P[k])) != 0:
+                x -= f[k] * np.log(np.trace(np.dot(rho, P[k])))
+        return x
+
+    def _grad_log_likelihood_func(self, rho: np.ndarray) -> float:
+        # Gradient of the log-likelihood function of the POVM
+
+        f = self._f
+        P = self._povm_operator()
+
+        grad = 0
+        for k in range(len(f)):
+            if np.trace(np.dot(rho, P[k])) != 0:
+                grad -= (f[k] / (np.trace(np.dot(rho, P[k])))) * P[k]
+        return grad
+
+    def state_tomography_density_matrix(self) -> np.ndarray:
+        return self._perform_mle_tomography(self._guess_density_matrix)
+
+    @staticmethod
+    def state_fidelity(target_state_dm: np.ndarray, computed_state_dm: np.ndarray) -> float:
+        """
+        Computes the fidelity of the density matrix reconstructed after State Tomography using MLE algorithm
+        :param target_state_dm: target Density Matrix of the State
+        :param computed_state_dm: reconstructed Density Matrix of the State
+        :return: fidelity of the reconstructed state
+        """
+        rx = sqrtm(target_state_dm)
+        z = np.linalg.multi_dot([rx, computed_state_dm, rx])
+        return np.real(np.trace(sqrtm(z)) ** 2)
+
+
+class ProcessTomographyMLE(TomographyMLE):
+    def __init__(self, operator_processor, **kwargs):
+        super().__init__(operator_processor, **kwargs)
+        self._f = self._povm_data()
+        self._guess_choi_seed = np.eye((2 ** self._nqubit), dtype='complex_')
+        self._guess_choi_matrix = np.kron(self._guess_choi_seed, self._guess_choi_seed) / 16
+    # Todo : Discuss and fix achritecture for fidelity and chi_matrix (Abstract class?)
+
+    def _povm_data(self) -> list:
+        # Performing a POVM (positive operator value measure) on the quantum processor
+        # in the informationally complete Pauli basis, i.e. choosing all the eigenvectors of the Pauli operators.
+        # They are |0>,|1>,|+>,|->,|i+>,|i->
+
+        # measurement is always on 3 Paulitype I, X, Y : Z is moved away
+        # prep has 6 options -> 4 pauli and 2 other are some combo
+        # of that itself -> find which and decide how to implement
+
+        f = []
+        preparation_states = _generate_pauli_prep_index(self._nqubit)
+        measurement_states = _generate_pauli_index(self._nqubit)
+
+        for index, value in enumerate(preparation_states):
```

Comment:
index is not use, you can remove enumerate

Reply:
I think this also remained from my debug, thanks for noticing these tiny things I missed